Show currently selected site in site visibility editor screen

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->
Closes: #13159
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Adds a new section to the "Edit site visibility" screen displaying the currently selected site. Additionally, it filters out the currently selected site from the list. 

### Testing information
1. Log in with an account that has multiple sites
2. Click `Edit Stores` button 
3. Check the currently selected site is displayed on the header
4. Check the currently selected site is not displayed in the editable list of sites

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
The above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="300"  src="https://github.com/user-attachments/assets/bafae305-ed85-4881-b430-cffc767d0802"> 

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Update UI to support displaying currently selected site

JorgeMucientes committed:
Show currently selected store and exclude it from the selectable list

JorgeMucientes committed:
Fix minor UI glitches

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Convert screen sections cards for more consistent UI

JorgeMucientes committed:
Convert screen sections cards for more consistent UI

JorgeMucientes committed:
Fix minor UI glitches

JorgeMucientes committed:
Fix unit tests

JorgeMucientes committed:
Fix lint warnings

irfano commented on WooCommerce/src/test/kotlin/com/woocommerce/android/ui/sitepicker/sitevisibility/WooSitesVisibilityViewModelTest.kt:
```diff
@@ -61,9 +65,8 @@ class WooSitesVisibilityViewModelTest : BaseUnitTest() {
             val updatedState = viewModel.viewState.getOrAwaitValue()
-            assert(updatedState?.wooStores?.first()?.isSelected == true)
-            assert(updatedState?.isSaveButtonEnabled == false)
+            assert(updatedState.wooStores?.first()?.isSelected == true)
```
We can simplify this line as you did in the previous function:

```suggestion
            assert(updatedState.wooStores.first().isSelected)
```

JorgeMucientes replied:
I fixed this already in a subsequent [PR](https://github.com/woocommerce/woocommerce-android/pull/13188), so I won't update it here to avoid unnecessary commits! But thank for the suggested change :)

irfano reviewed (changes requested):
The design and code look good. However, I noticed some issues:
- When the list is too long, the footer is not visible. 
<img src="https://github.com/user-attachments/assets/565869c6-7d01-42a4-8e2b-e12a3c265475" width=270>

- Perhaps we should consider making the entire screen scrollable instead of just the “Other Sites” section, as the current experience in landscape orientation is not ideal.

<img src="https://github.com/user-attachments/assets/a1aeb731-7542-4314-92f2-2eb9fbe27ac6" width=600>

JorgeMucientes committed:
Move descriptions under the title

JorgeMucientes committed:
Make the full screen scrollable

JorgeMucientes committed:
Remove testing code

JorgeMucientes committed:
Fill max size to keep the same background color

JorgeMucientes commented:
Thanks for the review @irfano! Good points!

I fixed both of them. To do that I made the whole screen scrollable and added the footer as subtitle so it is always visible: 

https://github.com/user-attachments/assets/e8571f02-8a22-4ad7-b5a7-cc120c5462ac

The alternative approach was to anchor the footer to the bottom, but since it is a very small text it didn't look well. I did a quick test (without adding the paddings or enything) and wasn't convinced about it. Let me know what you think.

https://github.com/user-attachments/assets/700322c5-3271-4860-b877-4f1cd8f9e200

irfano commented:
I don’t have a strong opinion, but placing the text at the top looks good. cc @itsmeichigo
